src/transformers/modeling_bert.py

```diff
@@ -852,9 +852,10 @@ def forward(
         return outputs  # (loss), prediction_scores, seq_relationship_score, (hidden_states), (attentions)
 
 
-# TODO: Split with a different BertWithLMHead to get rid of `lm_labels` here and in encoder_decoder.
-@add_start_docstrings("""Bert Model with a `language modeling` head on top. """, BERT_START_DOCSTRING)
-class BertForMaskedLM(BertPreTrainedModel):
+@add_start_docstrings(
+    """Bert Model with a `language modeling` head on top for CLM fine-tuning. """, BERT_START_DOCSTRING
+)
+class BertLMHeadModel(BertPreTrainedModel):
```

Comment:
I think the init should have an assert statement now to make sure `config.is_decoder=True` or even set `config.is_decoder=True` with a logging info that it does so. Using `BertLMHeadModel` without a causal mask does not make much sense IMO. 

Also, currently `BertLayer` always initializes the cross-attention layer if `config.is_decoder=True`, but doesn't use it if no `encoder_hidden_states` are passed to forward => so there is not really a problem atm when using the model on its own (not in an encoder-decoder framework). BUT, this should be corrected with an additional check `self.config.is_encoder_decoder`  before adding the cross-attention layers.

Reply:
Added an assert.